bcgov/biohubbc-platform
# BHBC-2224: (Backend + Types) See Dataset Documents

# Overview
As a Data Administrator or public user, I want to be able to see all the documents / file artifacts associated with a given dataset so I can download pieces of information that are of interest to me.
## Links to Jira tickets

 - https://quartech.atlassian.net/browse/BHBC-2276
 - https://quartech.atlassian.net/browse/BHBC-2277

## Description of relevant changes

 - Adds an endpoint for fetching artifacts by dataset ID
 - Adds a new frontend hook for fetching dataset attachments
 - Adds typedef for artifacts in the frontend
 - Adds service and repo methods for fetching artifacts by their respective dataset ID
 - Adds tests for new functions

## Testing Procedure

1. Create a project and survey in SIMS
2. Add one or more attachments to the survey
3. Click **Submit Data**
4. Click **SELECT ALL** and then submit.
5. In BioHub, navigate to the survey you submitted: **Map Search** -> **Species Inventory Project** -> **Find Data**
6. Open Inspect Element (Press **F12**)
7. Navigate to the **Network** tab and reload the page.
8. Click on the `attachments` request, then click **Preview**. You should:
  - [ ] See the attachment records
